core/src/wallet/cosmos/explorers/StargateGaiaCosmosLikeBlockchainExplorer.hpp

```diff
@@ -0,0 +1,198 @@
+    /// it will prevent the OperationQuery from fetching the correct block to compute the
+    /// number of confirmations for a given transaction.
+    /// \param[in] The transaction to fill
+    /// \return a FuturePtr to the filled Transaction
+    FuturePtr<cosmos::Transaction> inflateTransactionWithBlockData(const cosmos::Transaction& inputTx) const;
+
+    // Get all transactions relevant to an address
+    // Concatenates multiple API calls for all relevant transaction types
+    FuturePtr<cosmos::TransactionsBulk> getTransactionsForAddress(
+        const std::string &address, uint32_t fromBlockHeight = 0) const;
+
+    // Get all transactions relevant to a list of addresses
+    // Concatenates multiple API calls for all relevant transaction types
+    FuturePtr<cosmos::TransactionsBulk> getTransactionsForAddresses(
+        const std::vector<std::string> &addresses, uint32_t fromBlockHeight = 0) const;
+
+    Future<BigInt> genericPostRequestForSimulation(
+        const std::string &endpoint, const std::string &transaction) const;
+
+    Future<BigInt> getEstimatedGasLimit(
+        const std::shared_ptr<api::CosmosLikeTransaction> &transaction,
+        const std::shared_ptr<api::CosmosLikeMessage> &message,
+        const std::string gasAdjustment = "1.0") const;
+    Future<BigInt> getEstimatedGasLimitForTransfer(
+        const std::shared_ptr<api::CosmosLikeTransaction> &transaction,
+        const std::shared_ptr<api::CosmosLikeMessage> &message,
+        const std::string gasAdjustment = "1.0") const;
+    Future<BigInt> getEstimatedGasLimitForRewards(
+        const std::shared_ptr<api::CosmosLikeTransaction> &transaction,
+        const std::shared_ptr<api::CosmosLikeMessage> &message,
+        const std::string gasAdjustment = "1.0") const;
+    Future<BigInt> getEstimatedGasLimitForDelegations(
+        const std::shared_ptr<api::CosmosLikeTransaction> &transaction,
+        const std::shared_ptr<api::CosmosLikeMessage> &message,
+        const std::string gasAdjustment = "1.0") const;
+    Future<BigInt> getEstimatedGasLimitForUnbounding(
+        const std::shared_ptr<api::CosmosLikeTransaction> &transaction,
+        const std::shared_ptr<api::CosmosLikeMessage> &message,
+        const std::string gasAdjustment = "1.0") const;
+    Future<BigInt> getEstimatedGasLimitForRedelegations(
+        const std::shared_ptr<api::CosmosLikeTransaction> &transaction,
+        const std::shared_ptr<api::CosmosLikeMessage> &message,
+        const std::string gasAdjustment = "1.0") const;
+
+   private:
+    std::shared_ptr<HttpClient> _http;
+    api::Currency _currency;
+    /// GRPC namespace of the endpoints to use
+    std::string _ns{"cosmos"};
+    /// GRPC version of the endpoints to use
+    std::string _version{"v1beta1"};
+    /// Add a fees message to a transaction for internal purpose
+    template <typename T>
+    void addMsgFeesTo(cosmos::Transaction &transaction, const T &node) const;
```

Comment:
I put it in the private section because no one other than the explorer implementation should run this code, to avoid having multiple fees messages in transactions